bot-twitter/src/index.ts

```diff
@@ -1,21 +1,32 @@
+import NETWORKS from 'auctions-core/src/constants/NETWORKS';
+import { setSigner } from 'auctions-core/src/signer';
 import { getNewAuctions } from './auctions';
 import notify from './notify';
 
+const NETWORK = process.env.ETHEREUM_NETWORK || 'kovan';
 const DEFAULT_REFETCH_INTERVAL = 60 * 1000;
 const REFETCH_INTERVAL = parseInt(process.env.REFETCH_INTERVAL ?? '') || DEFAULT_REFETCH_INTERVAL;
 let refetchIntervalId: ReturnType<typeof setInterval> | undefined;
 
+if (!NETWORKS[NETWORK]) {
+    throw new Error(`network "${NETWORK}" is not supported`);
+}
+
+if (process.env.WALLET_PRIVATE_KEY) {
+    setSigner(NETWORK, process.env.WALLET_PRIVATE_KEY).then(() => {});
```

Comment:
- same as above
- if setSigner is async the error it throws will not be handled anywhere, so you need to await it to produce some kind of effect (ie stop program from executing). I suggest to define `setup` function and call seSigner there, right before initiating the loop. This way, the loop will not start if setSigner throws an error